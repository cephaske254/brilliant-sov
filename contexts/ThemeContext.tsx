import colorAlpha from "color-alpha";
import { createContext, ReactNode } from "react";
import theme from "../theme/theme";

const initialValues = {
  ...theme,
  alpha: colorAlpha,
};
const ThemeContext = createContext(initialValues);

const ThemeProvider = ({ children }: { children: ReactNode }) => {
  return (
    <ThemeContext.Provider value={initialValues}>
      {children}
    </ThemeContext.Provider>
  );
};


export { ThemeContext, ThemeProvider };

