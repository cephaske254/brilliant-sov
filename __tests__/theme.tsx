import renderer, { ReactTestRenderer } from "react-test-renderer";
import { ThemeProvider } from "../contexts/ThemeContext";
import Box from "../theme/Box";

test("Mock theme and theme provider", async () => {
  let wrapper: ReactTestRenderer;

  await renderer.act(() => {
    wrapper = renderer.create(
      <ThemeProvider>
        <Box></Box>
      </ThemeProvider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
