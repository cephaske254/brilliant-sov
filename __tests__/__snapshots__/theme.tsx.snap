// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock theme and theme provider 1`] = `null`;
